resources/auth.py

```diff
@@ -228,43 +195,37 @@ def post(self):
             description: |
                 User set password token and new password.
 
-                Password must be longer than 8 chars.
+                Password must be longer than 8 chars and shorter than 72 UTF-8 encoded bytes.
             content:
               application/json:
-                schema: UserSetPasswordSchema
+                schema: AuthSetPasswordRequestSchema
         responses:
             200:
               description: User login on successful password set
               content:
                 application/json:
-                  schema: UserSuccessSchema
+                  schema: UserSuccessResponseSchema
             400:
                 description: When request body is invalid or provided password doesn't match the policy
             403:
                 description: When set password token is no longer valid
         """
-        MIN_PASSWORD_LENGTH = 8
-        MAX_PASSWORD_LENGTH = 72  #UTF-8 bytes
-        schema = UserSetPasswordSchema()
+        schema = AuthSetPasswordRequestSchema()
         obj = schema.loads(request.get_data(as_text=True))
         if obj.errors:
             return {"errors": obj.errors}, 400
-        user = User.verify_set_password_token(obj.data.get("token"))
+
+        user = User.verify_set_password_token(obj.data["token"])
         if user is None:
             raise Forbidden("Set password token expired")
-        password = obj.data.get("password")
-        if password == "":
-            raise BadRequest("Empty password is not allowed")
-        if len(password) < MIN_PASSWORD_LENGTH:
-            raise BadRequest("Password is too short")
-        if len(password.encode()) > MAX_PASSWORD_LENGTH:
-            raise BadRequest("The password should contain no more than 72 bytes of UTF-8 characters, your password is too long.")
-
-        user.set_password(password)
-        db.session.add(user)
```

Comment:
This piece of logic was removed (well, it does look really weird, just making sure it's not a mistake).

Reply:
Definitely not a mistake, `session.add` was misused here.